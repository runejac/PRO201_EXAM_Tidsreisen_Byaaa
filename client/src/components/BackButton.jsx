import React from "react";
import "../css/backButton.css";
import { useNavigate } from "react-router-dom";
import { IoIosArrowBack } from "react-icons/io";

export function BackButton() {
  const navigate = useNavigate();
  return (
    <div id={"back-button-container"}>
      <IoIosArrowBack
        data-testid="backArrow"
        onClick={() => navigate(-1)}
        size={35}
      />
    </div>
  );
}
