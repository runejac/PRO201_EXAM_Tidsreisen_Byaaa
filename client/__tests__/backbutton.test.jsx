/**
 * @jest-environment jsdom
 */

import React from "react";
import { BackButton } from "../src/components/BackButton.jsx";
import { fireEvent, render } from "@testing-library/react";

const mockedUsedNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  useNavigate: () => mockedUsedNavigate,
}));

describe("BackButton", () => {
  it("shows snapshot", async () => {
    const component = render(<BackButton />);
    expect(component).toMatchSnapshot();
  });

  it("simulates click", async () => {
    const { getByTestId } = render(<BackButton />);
    fireEvent.click(getByTestId(/backArrow/i));
    expect(mockedUsedNavigate).toHaveBeenCalledTimes(1);
    expect(mockedUsedNavigate).toHaveBeenCalledWith(-1);
  });
});
