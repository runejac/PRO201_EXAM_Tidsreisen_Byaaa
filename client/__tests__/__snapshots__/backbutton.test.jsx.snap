// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton shows snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="back-button-container"
      >
        <svg
          data-testid="backArrow"
          fill="currentColor"
          height="35"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="35"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M217.9 256L345 129c9.4-9.4 9.4-24.6 0-33.9-9.4-9.4-24.6-9.3-34 0L167 239c-9.1 9.1-9.3 23.7-.7 33.1L310.9 417c4.7 4.7 10.9 7 17 7s12.3-2.3 17-7c9.4-9.4 9.4-24.6 0-33.9L217.9 256z"
          />
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      id="back-button-container"
    >
      <svg
        data-testid="backArrow"
        fill="currentColor"
        height="35"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="35"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M217.9 256L345 129c9.4-9.4 9.4-24.6 0-33.9-9.4-9.4-24.6-9.3-34 0L167 239c-9.1 9.1-9.3 23.7-.7 33.1L310.9 417c4.7 4.7 10.9 7 17 7s12.3-2.3 17-7c9.4-9.4 9.4-24.6 0-33.9L217.9 256z"
        />
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
